ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/iterating_linear_optimizer/IteratingLinearOptimizerWithLoopFlows.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.rao_commons.linear_optimisation.iterating_linear_optimizer;
+
+import com.farao_community.farao.rao_commons.CostEvaluator;
+import com.farao_community.farao.rao_commons.LoopFlowComputationService;
+import com.farao_community.farao.rao_commons.linear_optimisation.LinearOptimizer;
+import com.farao_community.farao.rao_commons.linear_optimisation.fillers.ProblemFiller;
+import com.farao_community.farao.rao_commons.SystematicSensitivityComputation;
+import com.farao_community.farao.util.SensitivityComputationException;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+/**
+ * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
+ */
+public class IteratingLinearOptimizerWithLoopFlows extends IteratingLinearOptimizer {
+
+    private boolean loopFlowApproximation;
+    private double loopFlowViolationCost;
+
+    public IteratingLinearOptimizerWithLoopFlows(List<ProblemFiller> fillers,
+                                                 SystematicSensitivityComputation systematicSensitivityComputation,
+                                                 CostEvaluator costEvaluator,
+                                                 IteratingLinearOptimizerWithLoopFLowsParameters parameters) {
+        super(fillers, systematicSensitivityComputation, costEvaluator, parameters);
+        loopFlowApproximation = parameters.isLoopflowApproximation();
+        loopFlowViolationCost = parameters.getLoopFlowViolationCost();
+        linearOptimizer = new LinearOptimizer(fillers);
+    }
+
+    @Override
+    protected boolean evaluateNewCost(String optimizedVariantId, int iteration) {
+        // If evaluating the new cost fails iteration can stop
+        raoData.setWorkingVariant(optimizedVariantId);
+        try {
+            LOGGER.info(format(SYSTEMATIC_SENSITIVITY_COMPUTATION_START, iteration));
+            runSensitivityAndUpdateResults();
+            Map<String, Double> loopFlows = LoopFlowComputationService.calculateLoopFlows(raoData, loopFlowApproximation);
```

Comment:
Here should have a test " is rao with loopflow"

Reply:
I think it is not necessary because this class is already IteratingLinearOptimizerWithLoopFlows so we check at the creation if we want to consider loop flows or not :
`if (raoParameters.isRaoWithLoopFlowLimitation()) {
    fillers.add(createMaxLoopFlowFiller(raoParameters));
    return new IteratingLinearOptimizerWithLoopFlows(fillers, systematicSensitivityComputation,
        createCostEvaluator(raoParameters), createIteratingLoopFlowsParameters(raoParameters));
} else {
    return new IteratingLinearOptimizer(fillers, systematicSensitivityComputation,
        createCostEvaluator(raoParameters), createIteratingParameters(raoParameters));
}`